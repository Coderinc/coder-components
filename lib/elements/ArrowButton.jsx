import PropTypes from 'prop-types';
import styled from 'styled-components';

const Button = styled.div`
  background-color: ${({
    disabled, secondary, theme, color,
  }) => {
    if (disabled) {
      return theme.offsetBackground;
    }

    if (secondary) {
      return theme.activeBackground;
    }
    return theme[color || 'primary'];
  }};

  color: ${({
    disabled, secondary, ternary, theme, color,
  }) => {
    if (disabled) {
      return theme.offsetFont;
    }

    if (ternary) {
      return theme.offsetFont;
    }
    if (secondary) {
      return theme[color || 'primary'];
    }
    return theme.primaryFont;
  }};

  outline: none;
  border: none;
  cursor: pointer;
  
  display: flex;
  justify-content: center;
  align-items: center;

  height: 2.5rem;
  width: 2.5rem;
  padding: 0.75rem;
  margin: 1rem;
  border-radius: 50%;
  box-sizing: border-box;

  box-shadow: ${({ disabled }) => (disabled ? undefined : '0 1px 3px rgba(0, 0, 0, 0.02), 0 1px 2px rgba(0, 0, 0, 0.04)')};
  transition: 0.5s ease all;

  &::before {
    content: '\f100';
    font-size: 0.875rem;
    font-family: coder-icons;
    transform: ${({ reversed }) => (reversed ? 'scaleX(-1)' : undefined)};
  }

  &:hover {
    box-shadow: ${({ disabled }) => (disabled ? undefined : '0 5px 10px rgba(0, 0, 0, 0.25), 0 5px 5px rgba(0, 0, 0, 0.10)')};
  }
`;

Button.propTypes = {
  disabled: PropTypes.bool,
  color: PropTypes.string,
  reversed: PropTypes.bool,
};

Button.defaultProps = {
  disabled: false,
  color: 'primary',
  reversed: false,
};

export default Button;
