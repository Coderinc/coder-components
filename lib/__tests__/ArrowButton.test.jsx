import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import sinon from 'sinon';
import 'jest-styled-components';
import ArrowButton from '../elements/ArrowButton';

describe('ArrowButton', () => {
  it('renders correctly', () => {
    const component = renderer.create(<ArrowButton />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('sets color correctly', () => {
    const component = renderer.create(<ArrowButton color="blue" theme={{ blue: '#0000FF' }} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('swaps button styles correctly', () => {
    const component = renderer.create(<ArrowButton color="blue" secondary theme={{ blue: '#0000FF' }} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('fire onClick', () => {
    const onClick = sinon.spy();
    const component = mount(<ArrowButton onClick={onClick} />);
    component.find('div').simulate('click');
    expect(onClick.calledOnce).toEqual(true);
  });
});
