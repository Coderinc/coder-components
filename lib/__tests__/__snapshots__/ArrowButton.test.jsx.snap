// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowButton renders correctly 1`] = `
.c0 {
  outline: none;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.5rem;
  width: 2.5rem;
  padding: 0.75rem;
  margin: 1rem;
  border-radius: 50%;
  box-sizing: border-box;
  box-shadow: 0 1px 3px rgba(0,0,0,0.02),0 1px 2px rgba(0,0,0,0.04);
  -webkit-transition: 0.5s ease all;
  transition: 0.5s ease all;
}

.c0::before {
  content: '\\f100';
  font-size: 0.875rem;
  font-family: coder-icons;
}

.c0:hover {
  box-shadow: 0 5px 10px rgba(0,0,0,0.25),0 5px 5px rgba(0,0,0,0.10);
}

<div
  className="c0"
/>
`;

exports[`ArrowButton sets color correctly 1`] = `
.c0 {
  background-color: #0000FF;
  outline: none;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.5rem;
  width: 2.5rem;
  padding: 0.75rem;
  margin: 1rem;
  border-radius: 50%;
  box-sizing: border-box;
  box-shadow: 0 1px 3px rgba(0,0,0,0.02),0 1px 2px rgba(0,0,0,0.04);
  -webkit-transition: 0.5s ease all;
  transition: 0.5s ease all;
}

.c0::before {
  content: '\\f100';
  font-size: 0.875rem;
  font-family: coder-icons;
}

.c0:hover {
  box-shadow: 0 5px 10px rgba(0,0,0,0.25),0 5px 5px rgba(0,0,0,0.10);
}

<div
  className="c0"
  color="blue"
/>
`;

exports[`ArrowButton swaps button styles correctly 1`] = `
.c0 {
  color: #0000FF;
  outline: none;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.5rem;
  width: 2.5rem;
  padding: 0.75rem;
  margin: 1rem;
  border-radius: 50%;
  box-sizing: border-box;
  box-shadow: 0 1px 3px rgba(0,0,0,0.02),0 1px 2px rgba(0,0,0,0.04);
  -webkit-transition: 0.5s ease all;
  transition: 0.5s ease all;
}

.c0::before {
  content: '\\f100';
  font-size: 0.875rem;
  font-family: coder-icons;
}

.c0:hover {
  box-shadow: 0 5px 10px rgba(0,0,0,0.25),0 5px 5px rgba(0,0,0,0.10);
}

<div
  className="c0"
  color="blue"
/>
`;
